great_expectations/render/renderer_configuration.py

```diff
@@ -0,0 +1,33 @@
+from dataclasses import dataclass, field
+from typing import Union
+
+from great_expectations.core import (
+    ExpectationConfiguration,
+    ExpectationValidationResult,
+)
+
+
+@dataclass(frozen=True)
+class RendererConfiguration:
+    """Configuration object built for each renderer."""
+
+    configuration: Union[ExpectationConfiguration, None]
+    result: Union[ExpectationValidationResult, None]
+    language: Union[str, None]
+    runtime_configuration: Union[dict, None]
+    include_column_name: Union[bool, None] = field(init=False)
+    styling: Union[dict, None] = field(init=False)
+
+    def __post_init__(self) -> None:
+        include_column_name: Union[bool, None] = None
+        styling: Union[dict, None] = None
+        if self.runtime_configuration:
+            include_column_name = (
+                False
+                if self.runtime_configuration.get("include_column_name") is False
+                else True
+            )
+            styling = self.runtime_configuration.get("styling")
+
+        object.__setattr__(self, "include_column_name", include_column_name)
+        object.__setattr__(self, "styling", styling)
```

Comment:
This pattern is what is used in `dataclass` constructors ([per the docs](https://docs.python.org/3/library/dataclasses.html#frozen-instances)) and is one of many approaches to setting frozen `dataclass` attributes that aren't allowed in the constructor (outlined [here](https://stackoverflow.com/questions/53756788/how-to-set-the-value-of-dataclass-field-in-post-init-when-frozen-true)).

Reply:
yup I agree! I don't love `setattr` but if this allows the object to stay frozen downstream, I'm all for it